nomenclature/codelist.py

```diff
@@ -149,6 +149,16 @@ def check_stray_tag(cls, v):
                 )
         return v
 
+    @validator("mapping")
+    def check_end_whitespace(cls, v):
+        """Check that no code ends with a whitespace"""
+        for code in v:
+            if code.endswith(" "):
+                raise ValueError(
+                    f"Unexpected white space at the end of a code: '{code}'."
```

Comment:
If you add `values` to the function call, you can make this error message more helpful
```suggestion
                    f"Unexpected whitespace at the end of a {values['name']} code: '{code}'."
```